shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java

```diff
@@ -497,7 +498,26 @@ private boolean isRestartAlwaysRequired() {
             mView.getContext().getContentResolver(), Settings.Secure.DEFAULT_INPUT_METHOD);
     // The Samsung keyboard is called "com.sec.android.inputmethod/.SamsungKeypad" but look
     // for "Samsung" just in case Samsung changes the name of the keyboard.
-    return keyboardName.contains("Samsung");
+    if (!keyboardName.contains("Samsung")) {
+      return false;
+    }
+
+    final long versionCode;
+    try {
+      versionCode =
+          mView
+              .getContext()
+              .getPackageManager()
+              .getPackageInfo("com.sec.android.inputmethod", 0)
+              .getLongVersionCode();
+    } catch (PackageManager.NameNotFoundException e) {
+      Log.w(TAG, "com.sec.android.inputmethod is not installed.");
+      return false;
+    }
+
+    // 3.3.31.83 is a known version that's free of the aforementioned bug.
+    // TODO(LongCatIsLooong): Find the minimum version that has the fix.
```

Comment:
Do you plan to do this in this PR or later?  Definitely my keyboard v2.0.01-44 has the bug, so it's between these two.

Reply:
Unfortunately I am still not able to reset the IME version. I've tried `adb uninstall --user 0`. It did uninstall the keyboard but it still wouldn't let me install an earlier version (says downgrade version).

@hamdikahloun thanks for the pointer! It's pretty handy.